### PR TITLE
Filter for date and datetime should always be a Date

### PR DESCRIPTION
**Pull request by megheaiulian**
Added serialize and deserialize for filter in date columns so that filter always contains Date objects.
For time column filter is string.

### CODE REVIEW

**Comment by plequang** [comment]
Not sure I understand @nomego 
> Actually, why we did this was to ensure consistent behavior.
> Timezones are only relevant when you have both a date and a time, not when you have only a date or only time.
The `formatter` is used to render the time to the user, not for comparison.  
Why would we use UTC to render the time ?
@megheaiulian fix seems legit to me.

**Comment by nomego** [comment]
Yes, I know. I just know that we went that road for a reason, I just can't recall the specifics right now.
But one thing is the `toISOString()` always returns UTC, and since we substring and ignore the timezone, this could mean that we input 22:00 but print out 20:00 due to timezone conversion between local and UTC.

Maybe @JockeCK knows more about the details?

But if you also add test-cases that confirm that whenever we print out a time, it is the same time as we inputted, then I'm happy.

**Comment by nomego** [comment]
Ok so this really boils down to `toISOString()` always being UTC.
```js
const f = new Intl.DateTimeFormat(undefined, {
					hour: 'numeric',
					minute: 'numeric',
					second: 'numeric'}),
  timeDate = new Date('1970-01-01T22:00');

f.format(timeDate); // "10:00:00 PM"
timeDate.toISOString().slice(11, 19); // "21:00:00"
```

Since we use them interchangeably, we either always need to honor timezone (never use `toISOString()`), or never (just stick to UTC).
Since a timestamp itself isn't associated with a timezone (unless it also has a date part) we have no need to honor timezone, so I'd say stick to UTC.

What we could do is to move all `new Date(this._fixedDate + 'T' + timestring)` stuff into a `_timeToUTCDate(timestring)` helper method, and in there actually add the UTC timezone (`return new Date(this._fixedDate + 'T' + timestring + 'Z')` instead of configuring the formatter to be UTC.

Not sure what the difference would be really, though. Neater?

@plequang @megheaiulian

**Comment by nomego** [comment]
Ok so after discussion with @megheaiulian we rather take other approach and be more timezone-aware.
This huge refactor also contains changes to range behavior, which has the potential of solving a few other issues as well.
We should review and squash-merge this as soon as possible if we want this in before next week and test it. (Squash so we can revert) Or we hold it off until after, your call @plequang

File: cosmoz-omnitable-column-date-behavior.html [hunk 1]
```diff
@@ -105,191 +45,84 @@
...
-				return new Date(value);
-			}
-			return value;
+			return this.renderValue(value, formatter);
 		},
 
-		getFilterFn(filter = this.filter) {
-			if (!filter || filter.start == null  && filter.end == null) {
+		renderValue(value, formatter = this.formatter) {
+			const date = this.toValue(value);
```
[the remark this answers is not in the document]
Comment (megheaiulian) [reply cosmoz-omnitable-column-date-behavior.html]: @plequang `renderValue` is also called from computeFilterText.

**Comment by megheaiulian** [comment]
@plequang I am aware of that issue but I am unsure how _limiting should work. Can you have a look at `_limitInput` in RangeColumnBehavior? I am simply setting _filterInput again to values within `_limit` range but I think we should do that depending on what changed. e.g.: If min change limit only it.

**Comment by megheaiulian** [comment]
@plequang I have pushed a fix for the limiting. Please have a look.

**Pull request merged**